api/organisations/tasks.py

```diff
@@ -189,7 +197,81 @@ def handle_api_usage_notifications():
             )
 
 
+def restrict_use_due_to_api_limit_grace_period_over() -> None:
+    grace_period = timezone.now() - timedelta(days=API_USAGE_GRACE_PERIOD)
+    month_start = timezone.now() - timedelta(30)
+    queryset = (
+        OranisationAPIUsageNotification.objects.filter(
+            notified_at__gt=month_start,
+            notified_at__lt=grace_period,
+            percent_usage__gte=100,
+        )
+        .values("organisation")
+        .annotate(max_value=Max("percent_usage"))
+    )
+
+    organisation_ids = []
+    for result in queryset:
+        organisation_ids.append(result["organisation"])
+    organisations = Organisation.objects.filter(
+        id__in=organisation_ids,
+        subscription__plan=FREE_PLAN_ID,
+        api_limit_access_block__isnull=True,
+    ).exclude(
+        stop_serving_flags=True,
+        block_access_to_admin=True,
+    )
+
+    organisation_ids = []
+    api_limit_access_blocks = []
+    for organisation in organisations:
+        api_limit_access_blocks.append(APILimitAccessBlock(organisation=organisation))
+        organisation_ids.append(organisation.id)
+
+    APILimitAccessBlock.objects.bulk_create(api_limit_access_blocks)
+
+    # Due to a bug in Django, calling update on the organisations
+    # queryset after it has been iterated through fails to save.
+    # So to work around that, we reload the queryset from the ids.
+    Organisation.objects.filter(
+        id__in=organisation_ids,
+    ).update(stop_serving_flags=True, block_access_to_admin=True)
+
+
+def unrestrict_after_api_limit_grace_period_is_stale() -> None:
```

Comment:
(We discussed during standup but for the benefit of others reading I'll reexplain it here.)

Once an account has entered a new month *and* they no longer have recent API usage notifications they're marked to be re-enabled and added back in. They can again be blocked if they're account usage puts them over the threshold again, but in the mean time they can serve flags and access the admin.

As for the flag being used on this task as well, I don't really see the usecase since this task is unsetting a punishment and I don't think we want to incorporate flags to stop the relief action from happening.